src/iarc7_motion/iarc_tasks/velocity_test_task.py

```diff
@@ -0,0 +1,128 @@
+from height_settings_checker import HeightSettingsChecker
+from acceleration_limiter import AccerlationLimiter
+
+class VelocityTestTaskState:
+    init = 0
+    moving = 1
+    waiting = 2
+
+class VelocityTestTask(AbstractTask):
+
+    def __init__(self, task_request):
+
+        self._drone_odometry = None
+        self._canceled = False
+
+        self._tf_buffer = tf2_ros.Buffer()
+        self._tf_listener = tf2_ros.TransformListener(self._tf_buffer)  
+
+        self._lock = threading.RLock()
+
+        self._current_velocity_sub = rospy.Subscriber(
+            '/odometry/filtered', Odometry,
+            self._current_velocity_callback)
+
+        try:
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MAX_TRANSLATION_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_Z_VELOCITY = rospy.get_param('~max_z_velocity')
+            self._TRANSLATION_SPEED_X = rospy.get_param('~test_velocity_x')
+            self._TRANSLATION_SPEED_Y = rospy.get_param('~test_velocity_y')
+
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for constant velocity test task')
+            raise
+
+        self._z_holder = HeightHolder(True)
+        self._height_checker = HeightSettingsChecker()
+        self._limiter = AccerlationLimiter()
+
+        self._state = VelocityTestTaskState.init
+
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
+    def get_desired_command(self):
+        with self._lock:
+            if (self._state == VelocityTestTaskState.init):
+                if self._drone_odometry is None:
+                    self._state = VelocityTestTaskState.waiting
+                else:
+                    self._state = VelocityTestTaskState.moving
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == VelocityTestTaskState.waiting):
+                if self._drone_odometry is None:
+                    self._state = VelocityTestTaskState.waiting
+                else:
+                    self._state = VelocityTestTaskState.moving
+                return (TaskRunning(), NopCommand())
+
+            elif self._canceled:
+                return (TaskCanceled(),)
+
+            elif self._state == VelocityTestTaskState.moving:
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+
+                x_vel_target = self._TRANSLATION_SPEED_X
+                y_vel_target = self._TRANSLATION_SPEED_Y
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
```

Comment:
Consider using copysign here for readability? https://stackoverflow.com/questions/1986152/why-doesnt-python-have-a-sign-function/1986776#1986776